src/com/intellij/idea/plugin/hybris/system/java/codeInsight/hints/LoggerInlayHintsProvider.kt

```diff
@@ -0,0 +1,178 @@
+    }
+
+    private inner class ClickHandler(
+        element: PsiElement,
+        private val loggerIdentifier: String,
+    ) : (MouseEvent?, Editor) -> Unit {
+        private val elementPointer = SmartPointerManager.createPointer(element)
+
+        override fun invoke(event: MouseEvent?, editor: Editor) {
+            if (isInlaySettingsEditor(editor)) return
+            val element = elementPointer.element ?: return
+            handleClick(editor, element, loggerIdentifier)
+        }
+    }
+
+    fun handleClick(editor: Editor, element: PsiElement, loggerIdentifier: String) {
+        val actionGroup = DefaultActionGroup().apply {
+            add(LoggerAction("TRACE", loggerIdentifier, HybrisIcons.Log.Level.TRACE))
+            add(LoggerAction("DEBUG", loggerIdentifier, HybrisIcons.Log.Level.DEBUG))
+            add(LoggerAction("INFO", loggerIdentifier, HybrisIcons.Log.Level.INFO))
+            add(LoggerAction("WARN", loggerIdentifier, HybrisIcons.Log.Level.WARN))
+            add(LoggerAction("ERROR", loggerIdentifier, HybrisIcons.Log.Level.ERROR))
+            add(LoggerAction("FATAL", loggerIdentifier, HybrisIcons.Log.Level.FATAL))
+            add(LoggerAction("SEVERE", loggerIdentifier, HybrisIcons.Log.Level.SEVERE))
+        }
+
+        val dataManager = DataManager.getInstance()
+
+        // Show the popup menu
+        val popup = JBPopupFactory.getInstance()
+            .createActionGroupPopup(
+                "Select an Option",
+                actionGroup,
+                dataManager.getDataContext(editor.component),
+                JBPopupFactory.ActionSelectionAid.SPEEDSEARCH,
+                true
+            )
+
+        // Calculate the position for the popup
+        val offset = element.endOffset
+        val logicalPosition = editor.offsetToLogicalPosition(offset)
+        val visualPosition = editor.logicalToVisualPosition(logicalPosition)
+        val point = editor.visualPositionToXY(visualPosition)
+
+        // Convert the point to a RelativePoint
+        val relativePoint = RelativePoint(editor.contentComponent, Point(point))
+
+        // Show the popup at the calculated relative point
+        popup.show(relativePoint)
+    }
+}
+
+class LoggerAction(private val logLevel: String, val logIdentifier: String, val icon: Icon) : AnAction(logLevel, "", icon) {
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project ?: return
+
+        val result = HybrisHacHttpClient.getInstance(project).executeLogUpdate(
+            project,
+            logIdentifier,
+            logLevel,
+            AbstractHybrisHacHttpClient.DEFAULT_HAC_TIMEOUT
+        )
+
+        val resultMessage = if (result.statusCode == 200) "Success" else "Failed"
+        val title = "Logger - $resultMessage"
+        Notifications.create(
+            NotificationType.INFORMATION, title,
+            "Set the log level: $logLevel for $logIdentifier. Server response: ${result.statusCode}"
```

Comment:
better wording may improve understanding, kinda: "Log level set to **FATAL** for server _server alias_."